### PR TITLE
kube-prometheus: Fix Alertmanager secret namespace.

### LINKED ISSUE #2017: Namespace in example.jsonnet not making into `alertmanager-secret.yaml`
**What did you do?**
* Create new dir
* `jb init`
* `jb install github.com/coreos/prometheus-operator/contrib/kube-prometheus/jsonnet/kube-prometheus`
* create `example.jsonnet` as the following
  ```
  > cat > example.jsonnet <<EOF
  local kp = (import 'kube-prometheus/kube-prometheus.libsonnet') + {
    _config+:: {
       namespace: 'monitoring',
    },
  };
  
  { ['00namespace-' + name]: kp.kubePrometheus[name] for name in std.objectFields(kp.kubePrometheus) } +
  { ['0prometheus-operator-' + name]: kp.prometheusOperator[name] for name in std.objectFields(kp.prometheusOperator) } +
  { ['node-exporter-' + name]: kp.nodeExporter[name] for name in std.objectFields(kp.nodeExporter) } +
  { ['kube-state-metrics-' + name]: kp.kubeStateMetrics[name] for name in std.objectFields(kp.kubeStateMetrics) } +
  { ['alertmanager-' + name]: kp.alertmanager[name] for name in std.objectFields(kp.alertmanager) } +
  { ['prometheus-' + name]: kp.prometheus[name] for name in std.objectFields(kp.prometheus) } +
  { ['grafana-' + name]: kp.grafana[name] for name in std.objectFields(kp.grafana) }
  EOF
  ```
* `mkdir manifests`
* `jsonnet -J vendor -m manifests example.jsonnet | xargs -I{} sh -c 'cat {} | gojsontoyaml > {}.yaml; rm -f {}' -- {}`

**What did you expect to see?**
```
> cat manifests/alertmanager-secret.yaml
apiVersion: v1
data:
  alertmanager.yaml: BLAHBLAHBLAH==
kind: Secret
metadata:
  name: alertmanager-main
  namespace: monitoring
type: Opaque
```
**What did you see instead? Under which circumstances?**
```
> cat manifests/alertmanager-secret.yaml
apiVersion: v1
data:
  alertmanager.yaml: BLAHBLAHBLAH==
kind: Secret
metadata:
  name: alertmanager-main
type: Opaque
```

This was working as expected about a month ago.

**Environment**
``` 
> jsonnet -v
Jsonnet commandline interpreter v0.11.2
```
`gotojson` and `jb` don't have such flags, but I did go get them.
OsX 10.13.6
* Prometheus Operator version:

	insert image tag or Git SHA here

* Kubernetes version information:
```
> kubectl version
Client Version: version.Info{Major:"1", Minor:"10", GitVersion:"v1.10.1", GitCommit:"d4ab47518836c750f9949b9e0d387f20fb92260b", GitTreeState:"clean", BuildDate:"2018-04-27T22:11:20Z", GoVersion:"go1.9.5", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"10+", GitVersion:"v1.10.3-eks", GitCommit:"58c199a59046dbf0a13a387d3491a39213be53df", GitTreeState:"clean", BuildDate:"2018-09-21T21:00:04Z", GoVersion:"go1.9.3", Compiler:"gc", Platform:"linux/amd64"}
```

* Kubernetes cluster kind: 
EKS v2

* Manifests:
See above
* Prometheus Operator Logs:
N/A

### PR DESCRIPTION
**Pull request by brancz**
Fixes #2017 

@metacyclic

### CODE REVIEW

File: contrib/kube-prometheus/jsonnetfile.lock.json [hunk 1]
```diff
@@ -78,7 +78,7 @@
                     "subdir": "Documentation/etcd-mixin"
                 }
             },
-            "version": "ba606bf85edfb3007f27a97d41f54e3fe3f70ce6"
+            "version": "7a759c18d294698f537f8be91927354818a71e51"
         }
     ]
-}
+}
```
Comment (squat) [inline contrib/kube-prometheus/jsonnetfile.lock.json]: @brancz are you running an old version of jb? I thought that new versions introduced a newline at the end of the lock file

File: contrib/kube-prometheus/jsonnet/kube-prometheus/alertmanager/alertmanager.libsonnet [hunk 2]
```diff
@@ -47,12 +47,12 @@ local k = import 'ksonnet/ksonnet.beta.3/k.libsonnet';
     secret:
       local secret = k.core.v1.secret;
 
-      if std.type($._config.alertmanager.config) == "object" then
-        secret.new('alertmanager-' + $._config.alertmanager.name, { 'alertmanager.yaml': std.base64(std.manifestYamlDoc($._config.alertmanager.config)) })
+      if std.type($._config.alertmanager.config) == 'object' then
+        secret.new('alertmanager-' + $._config.alertmanager.name, { 'alertmanager.yaml': std.base64(std.manifestYamlDoc($._config.alertmanager.config)) }) +
+        secret.mixin.metadata.withNamespace($._config.namespace)
```
Comment (metacyclic) [inline contrib/kube-prometheus/jsonnet/kube-prometheus/alertmanager/alertmanager.libsonnet]: Do you not want the trailing `,` here?

Comment (brancz) [reply contrib/kube-prometheus/jsonnet/kube-prometheus/alertmanager/alertmanager.libsonnet]: Actually jsonnet does not allow it here, it's not actually part of the if/else at the end, but rather ending the whole if/else.

File: contrib/kube-prometheus/jsonnetfile.lock.json (hunk 1 above)
Comment (brancz) [reply contrib/kube-prometheus/jsonnetfile.lock.json]: You're totally right, re-running.

File: contrib/kube-prometheus/jsonnet/kube-prometheus/alertmanager/alertmanager.libsonnet (hunk 2 above)
Comment (metacyclic) [reply contrib/kube-prometheus/jsonnet/kube-prometheus/alertmanager/alertmanager.libsonnet]: Sorry, I saw it on the similar line in the else block ...

**Pull request merged**